client/db.go

```diff
@@ -120,6 +120,17 @@ type Store interface {
 	// A lens configuration may also be provided, it will be added to all collections using the schema.
 	PatchSchema(context.Context, string, immutable.Option[model.Lens], bool) error
 
+	// PatchCollection takes the given JSON patch string and applies it to the set of CollectionDescriptions
+	// present in the database.
+	//
+	// It will also update the GQL types used by the query system. It will error and not apply any of the
+	// requested, valid updates should the net result of the patch result in an invalid state.  The
```

Comment:
todo: This sentence reads very odd to me "It will error and not apply any of the requested, valid updates should the net result of the patch result in an invalid state."

Please reword.

Reply:
What is odd about it?  Do you have a rough alternative in mind?

I think it is important to highlight that the full patch will be rolled back if an given patch item fails - especially given that SQLs do not do this and require manual rollback when given multiple DDL statements.

Reply:
This sounds more clear when you said: `full patch will be rolled back if an given patch item fails` 

This is very wordy: `... apply any of the requested, valid updates should the net resulted`

Reply:
`rolled back` is incorrect/lazy lol - the patches are aggregated before being applied - hence the use of `net result of the patch`

Reply:
> the patches are aggregated before being applied

This will be nice to document somewhere too

Reply:
> I though I understood the bahaviour from the prior sentence but this one gets me confused. How can the net result be valid if an individual operation creates an invalid state?

There are many reasons why we want this to be, and many of the tests cover this, for example:

> { "op": "copy", "from": "/Users/Fields/1", "path": "/Users/Fields/2" },
{ "op": "replace", "path": "/Users/Fields/2/Name", "value": "age" },

The first field is copied (invalid as it is now a duplicate), and then renamed (net is now valid).  This allows existing stuff (including full schema/collections) to be used as templates.  There are other use cases too.

> This will be nice to document somewhere too

It is, we are talking about the documentation that currently documents it.